apps/hpc-ftsadmin/src/app/pages/add-edit/flow-edit.tsx

```diff
@@ -0,0 +1,1857 @@
+import {
+  getSearchKeyValues,
+  getFormValueFromCategory,
+  getFormValueFromFunding,
+  getNameOfFundingValue,
+  FundingSrcType,
+  FundingObjectType,
+  compareSelectValues,
+  IndividualFormValueType,
+  isArrayFieldOfInputEntries,
+  enumInputEntryVsSrcObjType,
+} from '../../utils/parse-form-values';
+
+dayjs.extend(advancedFormat);
+
+interface Props {
+  className?: string;
+  isEdit: boolean;
+}
+
+interface Participant {
+  name?: string;
+  email?: string;
+}
+
+const StyledLoader = tw(C.Loader)`
+mx-auto
+`;
+
+const StyledAnchor = tw.a`
+underline
+`;
+
+const StyledAnchorDiv = tw.div`
+text-2xl
+float-right
+`;
+
+const StyledSubTitle = tw.h3`
+text-2xl
+`;
+
+const StyledInactiveTitle = tw.span`
+text-gray-500
+`;
+
+const StyledTitle = tw.div`
+flex
+justify-between
+`;
+const StyledRestrictedCheckBox = tw.div`
+inline-flex
+items-center
+justify-center
+`;
+
+const YellowTooltip = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} classes={{ popper: className }} />
+))(({ theme }) => ({
+  [`& .${tooltipClasses.tooltip}`]: {
+    backgroundColor: '#ffec1a',
+    color: 'rgba(0, 0, 0, 0.87)',
+    boxShadow: theme.shadows[1],
+    fontSize: 11,
+  },
+}));
+
+const initialFormData = {
+  amountUSD: '',
+  amountOriginal: '',
+  exchangeRateUsed: '',
+  keywords: [],
+  flowStatus: '',
+  flowType: { value: 133, displayLabel: 'Standard' },
+  flowDescription: '',
+  firstReported: dayjs().format('DD/MM/YYYY'),
+  decisionDate: null,
+  budgetYear: '',
+  flowDate: null,
+  contributionType: { value: 50, displayLabel: 'Financial' },
+  earmarkingType: '',
+  method: { value: 156, displayLabel: 'Traditional aid' },
+  beneficiaryGroup: '',
+  inactiveReason: '',
+  notes: '',
+  sourceOrganizations: [],
+  sourceLocations: [],
+  sourceUsageYears: [],
+  sourceProjects: [],
+  sourcePlans: [],
+  sourceGoverningEntities: [],
+  sourceGlobalClusters: [],
+  sourceEmergencies: [],
+  destinationOrganizations: [],
+  destinationLocations: [],
+  destinationUsageYears: [],
+  destinationProjects: [],
+  destinationPlans: [],
+  destinationGoverningEntities: [],
+  destinationGlobalClusters: [],
+  destinationEmergencies: [],
+  origCurrency: '',
+  includeChildrenOfParkedFlows: true,
+  reportDetails: [
+    {
+      verified: 'verified',
+      reportSource: 'primary',
+      reporterReferenceCode: '',
+      reportChannel: '',
+      reportedOrganization: { value: '', displayLabel: '' },
+      reportedDate: null,
+      reporterContactInformation: '',
+      sourceSystemRecordId: '',
+      reportFiles: [
+        {
+          title: '',
+          fileName: '',
+          UploadFileUrl: '',
+          size: 0,
+          type: '',
+        },
+      ],
+      reportFileTitle: '',
+      reportUrlTitle: '',
+      reportUrl: '',
+    },
+  ],
+  parentFlow: [],
+  childFlow: [],
+  isParkedParent: false,
+  sources: {},
+};
+
+const initialInputEntries = {
```

Comment:
Type initialInputEntries explicitly. TypeScript can't infer the types of empty arrays and many more.
```suggestion
const initialInputEntries: InputEntriesType = {
```